scripts/system/create/entitySelectionTool/entitySelectionTool.js

```diff
@@ -749,216 +749,463 @@ SelectionDisplay = (function() {
+        type: "Shape",
         alpha: 1,
         shape: "Cone",
-        solid: true,
+        primitiveMode: "solid",
         visible: false,
         ignorePickIntersection: true,
-        drawInFront: true
+        renderLayer: "front"
     };
     var handlePropertiesTranslateArrowCylinders = {
+        type: "Shape",
         alpha: 1,
         shape: "Cylinder",
-        solid: true,
+        primitiveMode: "solid",
         visible: false,
         ignorePickIntersection: true,
-        drawInFront: true
-    };
-    var handleTranslateXCone = Overlays.addOverlay("shape", handlePropertiesTranslateArrowCones);
-    var handleTranslateXCylinder = Overlays.addOverlay("shape", handlePropertiesTranslateArrowCylinders);
-    Overlays.editOverlay(handleTranslateXCone, { color: COLOR_RED });
-    Overlays.editOverlay(handleTranslateXCylinder, { color: COLOR_RED });
-    var handleTranslateYCone = Overlays.addOverlay("shape", handlePropertiesTranslateArrowCones);
-    var handleTranslateYCylinder = Overlays.addOverlay("shape", handlePropertiesTranslateArrowCylinders);
-    Overlays.editOverlay(handleTranslateYCone, { color: COLOR_GREEN });
-    Overlays.editOverlay(handleTranslateYCylinder, { color: COLOR_GREEN });
-    var handleTranslateZCone = Overlays.addOverlay("shape", handlePropertiesTranslateArrowCones);
-    var handleTranslateZCylinder = Overlays.addOverlay("shape", handlePropertiesTranslateArrowCylinders);
-    Overlays.editOverlay(handleTranslateZCone, { color: COLOR_BLUE });
-    Overlays.editOverlay(handleTranslateZCylinder, { color: COLOR_BLUE });
+        renderLayer: "front"
+    };
+    var handleTranslateXCone = Entities.addEntity(handlePropertiesTranslateArrowCones, "local");
+    var handleTranslateXCylinder = Entities.addEntity(handlePropertiesTranslateArrowCylinders, "local");
+    Entities.editEntity(handleTranslateXCone, { color: COLOR_RED });
+    Entities.editEntity(handleTranslateXCylinder, { color: COLOR_RED });
+    var handleTranslateYCone = Entities.addEntity(handlePropertiesTranslateArrowCones, "local");
+    var handleTranslateYCylinder = Entities.addEntity(handlePropertiesTranslateArrowCylinders, "local");
+    Entities.editEntity(handleTranslateYCone, { color: COLOR_GREEN });
+    Entities.editEntity(handleTranslateYCylinder, { color: COLOR_GREEN });
+    var handleTranslateZCone = Entities.addEntity(handlePropertiesTranslateArrowCones, "local");
+    var handleTranslateZCylinder = Entities.addEntity(handlePropertiesTranslateArrowCylinders, "local");
+    Entities.editEntity(handleTranslateZCone, { color: COLOR_BLUE });
+    Entities.editEntity(handleTranslateZCylinder, { color: COLOR_BLUE });
+
+    var materialHandleTranslateXCone = Entities.addEntity({
+        "type": "Material",
+        "parentID": handleTranslateXCone,
+        "name": "tool-material",
+        "parentMaterialName": "0",
+        "materialURL": "materialData",
+        "priority": 1,
+        "materialMappingMode": "projected",
```

Comment:
I don’t think this should be projected, it should use the other mode (probably doesn’t make a difference here but it makes the shaders do more calculations)

Reply:
also I think it might be beneficial to mark the materials as all ignorePickIntersection: true